### PR TITLE
Added an example endpoint for Data API

### PR DESCRIPTION
**Pull request by lloyddewit**
Closes #3 
I merged changes from PR #3 .

This PR adds an endpoint that calls the Data API as described in [https://github.com/opencdms/opencdms-process#readme](https://github.com/opencdms/opencdms-process#readme).

The end point requires the `opencdms-test-data` to be installed.
On Linux, I did this with:

```
cd ~
mkdir opencdms
cd opencdms
git clone https://github.com/opencdms/opencdms-test-data.git
```

I'm happy we have this example working. I'm not sure if we can use this data API for the demo. We can discuss with @isedwards.

### CODE REVIEW

**Review by chrismclarke — CHANGES_REQUESTED** [review]
I've merged #3 and updated to show the correct diffs.

If I understand this PR correctly it adds support for calling opencdms data functions by cloning the repo that contains them and placing in a subfolder? 

If that is the case, cool to have this piece of the puzzle, although in its current form I expect this will both break the cloud build and fail locally for people running without prior knowledge of how to include additional setup. In which case I will block for now so that we can either:

- import from a published repo and add to requirements (if possible - will probably need to check publication status with Ian)
OR
- add a note to the readme to include initial setup instructions
- add a line to DOCKERFILE to include the manual steps

Let me know your preferred approach @lloyddewit and if you need any support with either.

File: app/main.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 from base64 import b64decode
 from cmath import nan
+from pathlib import Path
 from typing import Dict, List, Optional
 from uuid import uuid4
 
 import uvicorn  # needed for debugging, see https://fastapi.tiangolo.com/tutorial/debugging/
 from fastapi import FastAPI
 from fastapi.responses import FileResponse
+from opencdms import MidasOpen
```
Comment (chrismclarke) [inline app/main.py:12]: Is this picked up from the cloned repo, or something that you have installed locally which needs to be added to the requirements file?

Comment (lloyddewit) [reply app/main.py:12]: Thanks, I added to `requirements.txt` using `pip3 freeze > requirements.txt`

**Comment by lloyddewit** [comment]
@chrismclarke Thank you for the review.
To keep this PR tidy, I updated the `requirements.txt` and `README.md` files.
It's probably best if you update `DOCKERFILE` (but it may not be needed, so feel free to wait until after we've spoken with @isedwards).

I'm not sure if we'll eventually include this endpoint in the demo so I set this PR to draft for now.

**Comment by lloyddewit** [comment]
@chrismclarke 
Just for your information: I uninstalled `backports.zoneinfo` with `pip3 uninstall backports.zoneinfo`.
But this triggered a runtime error:
```
File "/home/stephen/opencdms/opencdms-components-server-demo/.venv/lib/python3.8/site-packages/pytz_deprecation_shim/_compat_py3.py", line 7, in <module>   from backports import zoneinfo
ModuleNotFoundError: No module named 'backports'
```
So I reinstalled it.
It looks like this module is needed by Python 3.8. I'm running Python 3.8 to be compatible with the `opencdms-process` repo which only currently works with Python 3.8.
I may switch to Python 3.10 for `opencdms-components-server-demo`, but for now I'll just keep `backports.zoneinfo` installed locally.

**Review by chrismclarke — APPROVED** [review]
I'm going to go ahead and merge this so we can continue to build on top, we can always remove things later if no longer required.  Adding the requirement did indeed fix the docker build so no additions required to the dockerfile

I've added tests to confirm docker build passes and ran on a parallel branch (via #7) to allow access to secrets (which also gives the test pass here because of some weird github sha magic) - so I'd say all good

**Pull request merged**